pythran/pythonic/include/types/complex.hpp

```diff
@@ -39,6 +39,20 @@ struct __combined<std::complex<double>, indexable<K>> {
 
 /* } */
 
+#define STD_COMPLEX_IMPLICT_OPERATOR_CAST(op)                                  \
+  template <class T, class U>                                                  \
+  auto operator op(std::complex<T> const &lhs, std::complex<U> const &rhs)     \
+      ->std::complex<typename std::common_type<T, U>::type>                    \
+  {                                                                            \
+    using ctype = std::complex<typename std::common_type<T, U>::type>;         \
+    return ctype{lhs} + ctype{rhs};                                            \
+  }
+
+STD_COMPLEX_IMPLICT_OPERATOR_CAST(+)
+STD_COMPLEX_IMPLICT_OPERATOR_CAST(-)
+STD_COMPLEX_IMPLICT_OPERATOR_CAST(*)
+STD_COMPLEX_IMPLICT_OPERATOR_CAST(/ )
```

Comment:
ISTM the operator already exists when ``T`` == ``U``... how do you handle the conflict?

Without a test, it's hard to tell if this work, please add a few in ``pythran/tests/test_complex.py``